p8e-shared/src/main/kotlin/io/provenance/p8e/shared/service/AffiliateService.kt

```diff
@@ -53,17 +58,58 @@ class AffiliateService(
         const val AFFILIATE_ENCRYPTION_KEY_PAIR = "affiliate_encryption_key_pair"
         const val AFFILIATE_BECH32_LOOKUP = "affiliate_bech32_lookup"
         const val PUBLIC_KEY_TO_ADDRESS = "public_key_to_address"
+        const val AFFILIATE_SIGNING_KID = "affiliate_signing_key_id"
+        const val AFFILIATE_ENCRYPTION_PUBLIC_KEY = "affiliate_encryption_public_key"
+    }
+
+    /**
+     *  Return the signer to be used
+     *
+     *  [publicKey] publicKey used to validate
+     *  [signer] The specific signer used to sign contracts
+     */
+    fun getSigner(publicKey: PublicKey): SignerImpl {
+        val affiliateRecord = get(publicKey)
+        return if(affiliateRecord?.keyType == SMARTKEY) {
```

Comment:
one nice thing if this was a `when` instead of an `if/else` is that if we were to add a third type enum it would give compilation errors here saying it wasn't implemented in the `when`. As it currently is a third type that wasn't handled here would fall into the `else`.